### PR TITLE
Add support for installing using Nix

### LINKED ISSUE #70: Nix support
Adding a `default.nix` would help users on NixOS, or using Nix, to easily build himalaya. Here's the default.nix that I used to build the current master branch:

```nix
let 
  pkgs = import <nixpkgs> {};
in 
pkgs.rustPlatform.buildRustPackage rec {
  pname = "himalaya";
  version = "0.2.2";

  src = ./.;

  nativeBuildInputs = with pkgs; [
    pkg-config 
    openssl.dev
  ];
  PKG_CONFIG_PATH = "${pkgs.openssl.dev}/lib/pkgconfig";

  cargoSha256 = "sha256-i8YKE5Z2xoNyjFktRRVA45zD6Qa0aPNC0AOjHUIglgk=";

  meta = with pkgs.stdenv.lib; {
    description = "Minimalist CLI email client";
    homepage = "https://github.com/soywod/himalaya";
  };
}
```

### PR DESCRIPTION
**Pull request by srid**
[What's Nix?](https://serokell.io/blog/what-is-nix) It is the best way to [install](https://neuron.zettel.page/install) software in a *reproducible* and automated fashion!

Resolves #70 

- [x] Running `nix-build` produces a binary at `./result/`
- [x] Consider [flake](https://www.tweag.io/blog/2020-05-25-flakes/) support
- [x] Nix best practices
    - [x] Respect .gitignore
    - [x] Pin nixpkgs
- [x] Documentation
  - [x] Document Nix maintenance workflow
      - how to keep version in sync with Cargo.toml (automate?)
      - how to update the Cargo dependency sha (done manually when Cargo deps change)
  - [x] Add nix instructions to README

### CODE REVIEW

**Comment by srid** [comment]
The commit that I just pushed also makes this PR support Nix-based development. Basically by running `nix develop -c "code ."` (or `nix-shell --run "code ."` if Flakes is not being used) - you open VSCode with full Rust IDE support. Rust need not be installed on the system; the only requirement is that Nix is installed.

image

For folks who use Vim, then can simply launch their editor from inside the aforementioned nix-shell. 

This is also how I develop my Haskell projects. :-) https://github.com/srid/neuron/blob/master/CONTRIBUTING.md#developing

**Comment by srid** [comment]
@soywod It's all done! 

A couple of notes:

- As the README indicates, the SHA has will have to be updated (easy to do, if you have Nix installed) whenever the Cargo dependency tree changes. 
- I could've added GitHub Actions CI that runs the full nix build on every commit, but I refrained from doing so - because I do not know if you wish to or plan on using Nix for development (Nix does have a learning curve ahead, but I generally recommend it). But I'd be happy to open a separate PR if you're open to it. It would catch accidental forgetting of updating of the aforementioned SHA, and make sure that the Nix based install remains functional over time.

File: README.md [hunk 1]
```diff
@@ -9,28 +9,31 @@ Minimalist CLI email client, written in Rust.
...
+    - [Reply to a message](#reply-to-a-message)
+    - [Forward a message](#forward-a-message)
+    - [Copy a message](#copy-a-message)
+    - [Move a message](#move-a-message)
+    - [Delete a message](#delete-a-message)
+    - [Listen to new messages](#listen-to-new-messages)
+  - [Interfaces](#interfaces)
+    - [GUI](#gui)
+    - [TUI](#tui)
+  - [Credits](#credits)
```
Comment (srid) [inline README.md:36]: Huh, not sure what happened here. I didn't realize VSCode magically modified the TOC from behind the scenes.

Comment (srid) [reply README.md]: I had a VSCode extension that did this.

File: README.md [hunk 2]
```diff
@@ -49,9 +52,44 @@ more flexibility.
...
+If you already have Nix installed, run the following command to install himalaya:
+
+```bash
+nix-env -if https://github.com/soywod/himalaya/archive/master.tar.gz
+```
+
+If you use Nix with [Flakes](https://nixos.wiki/wiki/Flakes), you can directly run himalaya using:
+
+```bash
+nix run github:soywod/himalaya
```
Comment (srid) [inline README.md:73]: These two commands (`nix-env` and `nix run`) work only after merging this PR. But they can be tested by substituting for the GitHub user and branch name of my fork, viz.:

```bash
nix-env -if https://github.com/srid/himalaya/archive/nixify.tar.gz
```

and,

```bash
nix run github:srid/himalaya/nixify
```

image

**Comment by srid** [comment]
Just pushed a change that obviates having to maintain Nix files over time. No need to do anything when Cargo.lock or version changes.

**Comment by srid** [comment]
Something's broken after the "Improve Nix workflow, for zero maintenance" - let me address that first.

**Comment by srid** [comment]
Oh, well, n/m - I've gone back to the working state. This PR is ready for merging.

**Comment by soywod** [comment]
Thanks for this awesome contribution :heart:
Since I don't know enough about Nix, I plan to dive in this weekend to understand at least the concept. This way I will understand better what you did and I will be able to merge your work.

**Comment by srid** [comment]
This PR is now superseded by the improved #76, so I'll close it.

@soywod Sounds good to me. [Here's the reddit post](https://www.reddit.com/r/rust/comments/mmbfnj/nixifying_a_rust_project/) I made to try to understand the Nix story for Rust while working on this PR (which knowledge I'll borrow for other projects).

**Pull request closed without merging**